src/app/src/utils/providers/gitlab.ts

```diff
@@ -0,0 +1,168 @@
+import { ofetch } from 'ofetch'
+import { joinURL } from 'ufo'
+import type { GitOptions, GitProviderAPI, GitFile, RawFile, CommitResult, CommitFilesOptions } from '../../types'
+import { DraftStatus } from '../../types/draft'
+
+export function createGitLabProvider(options: GitOptions): GitProviderAPI {
+  const { owner, repo, token, branch, rootDir, authorName, authorEmail, instanceUrl = 'https://gitlab.com' } = options
+  const gitFiles: Record<string, GitFile> = {}
+
+  // GitLab uses project path (namespace/project) encoded as project ID
+  const projectPath = encodeURIComponent(`${owner}/${repo}`)
+  const baseURL = `${instanceUrl}/api/v4`
```

Comment:
I think we should use or pass the V4-URL directly because its pretty common for GitLab CI to use this variable when deploying.

<img width="766" height="71" alt="Image" src="https://github.com/user-attachments/assets/1f5c3b0d-f572-4b26-a061-4d133c8089cc" />

Reply:
@larbish I agree, you can do that 👌